tests/supervisor/test_supervisor.c

```diff
@@ -0,0 +1,65 @@
+#define _GNU_SOURCE
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <string.h>
+#include <inttypes.h>
+#include <unistd.h>
+#include <setjmp.h>
+#include <stdint.h>
+#include <cmocka.h>
+
+#include "utils/log.h"
+#include "supervisor/supervisor.h"
+#include "supervisor/sqlite_macconn_writer.h"
+
+static const UT_icd config_ifinfo_icd = {sizeof(config_ifinfo_t), NULL, NULL,
+                                         NULL};
+
+static void test_get_mac_conn_cmd(void **state) {
+  (void)state; /* unused */
+  uint8_t mac_addr[6] = {0x04, 0xf0, 0x21, 0x5a, 0xf4, 0xc4};
+  uint8_t wpa_passphrase[4] = {0x1, 0x2, 0x3, 0x0};
+
+  struct supervisor_context ctx = {};
```

Comment:
The ` = {}` syntax is only valid in C23, so it might not work on older compilers.

```suggestion
  struct supervisor_context ctx = {0};
```